benches/search_comparison.rs

```diff
@@ -285,3 +240,64 @@ fn make_sorted_vec<T: Ord>(mut v: Vec<T>) -> Vec<T> {
 fn search_sorted_vec<'a, T: Ord>(c: &'a Vec<T>, x: T) -> Option<&'a T> {
     c.binary_search(&x).ok().map(|i| &c[i])
 }
+
+fn test_sequence<T>(duplicates: bool, max: usize) -> impl Iterator<Item = T>
+where
+    T: TryFrom<usize>,
+    <T as TryFrom<usize>>::Error: core::fmt::Debug,
+{
+    let mut index = 0;
+    std::iter::from_fn(move || {
+        let item = if duplicates {
+            let item = std::cmp::min(index, max);
+            item % 256
+        } else {
+            std::cmp::min(index * 2, max)
```

Comment:
For `u8` the sequence will be:
```
0 2 4 6 .. 254 256 256 256 256
```
Which doesn't really make sense to call it `duplicates = false`?

Reply:
Indeed, we cannot generate more than 128 values for `u8` without duplicates (only even values). I think it is implemented in such a general way to get rid of duplicated code.